packages/runtime/src/slate/ColorPlugin/index.tsx

```diff
@@ -0,0 +1,61 @@
+import { Editor, Text } from 'slate'
+import {
+  Color,
+  createRichTextV2Plugin,
+  unstable_StyleV2,
+  ColorControlDefinition,
+} from '../../controls'
+import { ControlDefinitionValue } from '../../runtimes/react/controls/control'
+import {
+  getResponsiveValue,
+  normalizeResponsiveValue,
+  setResponsiveValue,
+} from '../utils/responsive'
+
+export const COLOR_KEY = 'color'
+
+export function ColorPlugin() {
+  return createRichTextV2Plugin({
+    control: {
+      definition: unstable_StyleV2({
+        type: Color({
+          label: 'Color',
+        }),
+        getStyle(color: ControlDefinitionValue<ColorControlDefinition>) {
+          return { color }
+        },
+      }),
+      onChange: (editor, value) => {
+        setResponsiveValue(editor, COLOR_KEY, value, {
+          match: Text.isText,
+          split: true,
+        })
+      },
+      getValue: editor => getResponsiveValue(editor, COLOR_KEY, { match: Text.isText }),
+      getLeafValue: (text: Text) => {
+        return Text.isText(text) ? text?.color : undefined
+      },
+    },
+    withPlugin: (editor: Editor) => {
+      const { normalizeNode } = editor
+      editor.normalizeNode = entry => {
+        if (normalizeResponsiveValue(editor, COLOR_KEY, { match: Text.isText })(entry)) {
```

Comment:
I don't think the final implementation of this will use a partial application, but it's fine for now. I'll be using these helpers in the other plugins so I'll have time to refine / add tests